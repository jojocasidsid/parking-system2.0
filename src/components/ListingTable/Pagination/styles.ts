import styled from '@emotion/styled'
import { Typography } from '@mui/material'

export interface IPageInputProps {
	error?: string
}

export const PageInput = styled.input<IPageInputProps>`
	border-radius: 2px 0px 0px 2px;
	padding: 7px 2px 7px 0px;
	text-align: center;
	width: 100%;
	font-style: normal;
	font-size: 14px;
	line-height: 16px;
	font-weight: 500;
	font-stretch: normal;
	height: 28px;

	&::-webkit-outer-spin-button,
	&::-webkit-inner-spin-button {
		-webkit-appearance: none;
		margin: 0;
	}

	&:active,
	&:focus {
		border: 1px solid ${({ theme }) => theme.palette.primary.main};
		outline: none;
		box-shadow: none;
	}

	&:disabled {
		background-color: ${({ theme }) => theme.palette.text.secondary};
		pointer-events: none;
	}

	border: 1px solid
		${({ theme, error }) => {
			if (error) {
				return theme?.palette.error.main
			}
			return theme.palette.text.secondary
		}};
`

export const StyledTypography = styled(Typography)`
	color: #303643b3;
	font-weight: bold;
	font-size: 12px;
`
