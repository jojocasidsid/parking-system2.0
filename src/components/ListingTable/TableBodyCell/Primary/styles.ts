import styled from '@emotion/styled'
import Typography from '@mui/material/Typography'

export const StyledTypography = styled(Typography)`
	color: #303643b3;
	font-weight: bold;
`
