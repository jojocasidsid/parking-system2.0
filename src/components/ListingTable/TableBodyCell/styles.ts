import styled from '@emotion/styled'
import Typography from '@mui/material/Typography'
import TableRow from '@mui/material/TableRow'

export const StyledTableRow = styled(TableRow)`
	cursor: pointer;
`
export const StyledTypography = styled(Typography)`
	color: #303643b3;
	font-weight: bold;
`
