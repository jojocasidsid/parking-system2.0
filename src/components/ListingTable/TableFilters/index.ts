import TableFilters from './TableFilters'

export default TableFilters
