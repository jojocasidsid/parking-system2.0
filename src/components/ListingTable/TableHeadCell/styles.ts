import styled from '@emotion/styled'
import Typography from '@mui/material/Typography'

export const StyledTableHeadTypography = styled(Typography)`
	font-weight: bold;
	font-size: 14px;
`
