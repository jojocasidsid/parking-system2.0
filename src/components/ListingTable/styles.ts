import styled from '@emotion/styled'
import { Typography } from '@mui/material'

export const StyledTypography = styled(Typography)`
	color: #30364380;
	font-weight: bold;
	font-size: 14px;
`
