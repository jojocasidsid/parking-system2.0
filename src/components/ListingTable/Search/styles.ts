import styled from '@emotion/styled'
import { InputLabel, TextField } from '@mui/material'
import { Theme } from '@mui/material/styles'

interface IInput {
	error?: boolean
	theme?: Theme
	small?: boolean
}

export const StyledRoot = styled('div')`
	display: block;
`

export const StyledLabel = styled(InputLabel)`
	text-align: 'left';
	letter-spacing: 0.12px;
	padding-bottom: 8px;

	font-style: normal;
	font-size: 12px;
	line-height: 14px;
	font-weight: 600;
	font-stretch: normal;
	white-space: normal;
	//has error
	color: ${({ theme, error }: IInput) => {
		if (error) {
			return theme?.palette.error.main
		}
		return theme?.palette.text.primary
	}};
	& span {
		color: ${({ theme }) => theme.palette.error.main};
	}

	display: block;
`

export const StyledInput = styled(TextField)`
	input {
		padding: ${({ small }: IInput) =>
			small ? '2px 8px 2px 8px' : '8px 12px 8px 16px'};
		font-style: normal;
		font-size: 14px;
		font-weight: 500;
		font-stretch: normal;
		color: ${({ theme }) => theme?.palette.text.primary};
	}

	fieldset {
		border: 1px solid ${({ theme }) => theme.palette.text.secondary};
		border-radius: 2px 0px 0px 2px;
	}

	input.Mui-disabled {
		background-color: ${({ theme }) => theme.palette.text.secondary};
		pointer-events: none;
	}

	input.MuiInputBase-inputAdornedStart {
		padding-left: 0px !important;
	}

	input.MuiInputBase-inputAdornedEnd {
		padding-right: 0px;
	}

	.MuiInputBase-adornedEnd div.MuiInputAdornment-root {
		background-color: ${({ theme }) => theme.palette.primary.contrastText};
		padding: 16px 16px 16px 16px;
		padding-right: 0px !important;
	}

	.MuiInputBase-adornedStart {
		padding-left: 0px !important;
	}

	.MuiInputBase-adornedStart .MuiInputAdornment-root {
		background-color: ${({ theme }) => theme.palette.primary.contrastText};
		padding: 16px 16px 16px 16px;
	}

	.MuiFormHelperText-root {
		margin-left: 0;
	}
`
