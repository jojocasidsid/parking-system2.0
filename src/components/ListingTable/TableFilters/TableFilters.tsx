import React from 'react'
import Grid from '@mui/material/Grid'
import Search from '../Search'
import { IFilters } from '../types'

interface IProps {
	filters?: IFilters
	setFilters?: React.Dispatch<React.SetStateAction<IFilters>>
}

const TableFilters = ({ filters, setFilters }: IProps) => (
	<Grid item xs='auto'>
		{filters && typeof filters.q !== 'undefined' && (
			<Search value={filters && filters.q} setValue={setFilters} />
		)}
	</Grid>
)

export default TableFilters
