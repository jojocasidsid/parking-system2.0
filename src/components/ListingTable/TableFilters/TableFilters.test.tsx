import React, { useState } from 'react'
import renderer from 'react-test-renderer'

// packages
import { ThemeProvider } from '@mui/material/styles'

// config
import theme from 'config/theme'

// component
import { IFilters } from 'hooks/useListingTable'
import TableFilters from './TableFilters'

describe('Table Filters  Component', () => {
	it('should render Table Filters component without crashing', () => {
		const Component = () => {
			const [filters, setFilters] = useState<IFilters>({
				_order: 'asc',
				_sort: '',
				_page: 1,
				_limit: 20,
				q: '',
			})

			return (
				<ThemeProvider theme={theme}>
					<TableFilters filters={filters && filters} setFilters={setFilters} />
				</ThemeProvider>
			)
		}

		const tree = renderer.create(<Component />)
		expect(tree).toMatchSnapshot()
	})
})
