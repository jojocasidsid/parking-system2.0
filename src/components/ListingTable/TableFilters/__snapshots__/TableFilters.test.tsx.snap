// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Filters  Component should render Table Filters component without crashing 1`] = `
<div
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1wrgmsj-MuiGrid-root"
>
  <div
    className="css-evn1e9"
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary css-uj5iz7-MuiFormLabel-root-MuiInputLabel-root"
      htmlFor="search"
    >
      Search
    </label>
    <div
      className="MuiFormControl-root MuiTextField-root css-1yfph09-MuiFormControl-root-MuiTextField-root"
    >
      <div
        className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1k534au-MuiInputBase-root-MuiOutlinedInput-root"
        disableUnderline={true}
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id="searchBar"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Search Transactions"
          required={false}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-hdw1oc"
          >
            <span
              className="notranslate"
            >
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;
